### ISSUE TITLE
Can't import android-google-play-services-ads 22.3.0

### ISSUE THREAD

**Issue by Bolfish** [dandar3/android-google-play-services-ads#7]
When I import : https://raw.githubusercontent.com/dandar3/android-google-play-services-ads/22.3.0/.projectset
The following errors occurs:

```
svn: E160013: '/dandar3/android-google-play-services-measurement-base/tags/20.1.2' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-cursoradapter/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-startup-runtime/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-arch-core-common/tags/2.1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-slidingpanelayout/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-measurement-sdk-api/tags/20.1.2' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-arch-core-runtime/tags/2.1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-lifecycle-livedata/tags/2.1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-core/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-legacy-support-core-ui/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-ump-user-messaging-platform/tags/2.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-lifecycle-service/tags/2.1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-collection/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-sqlite-framework/tags/2.1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-asynclayoutinflater/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-work-runtime/tags/2.7.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-ads-base/tags/22.3.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-legacy-support-core-utils/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-coordinatorlayout/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-fragment/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-loader/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-tasks/tags/18.0.1' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-tracing/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-interpolator/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-room-runtime/tags/2.2.5' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-appset/tags/16.0.1' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-swiperefreshlayout/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-ads-identifier/tags/18.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-customview/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-annotation/tags/1.1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-guava-listenablefuture/tags/1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-base/tags/18.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-versionedparcelable/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-lifecycle-runtime/tags/2.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-sqlite/tags/2.1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-print/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-ads-lite/tags/22.3.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-basement/tags/18.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-concurrent-futures/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-lifecycle-livedata-core/tags/2.1.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-annotation-experimental/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-lifecycle-viewmodel/tags/2.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-lifecycle-common/tags/2.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-documentfile/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-room-common/tags/2.2.5' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-viewpager/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-google-play-services-ads/tags/22.3.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-browser/tags/1.4.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-localbroadcastmanager/tags/1.0.0' path not found: 404 Not Found (https://github.com)
svn: E160013: '/dandar3/android-androidx-drawerlayout/tags/1.0.0' path not found: 404 Not Found (https://github.com)
```

**Comment by dandar3** [comment]
@Bolfish 
Thank you for reporting this. It appears GitHub.com has removed the support for SVN due to low usage / maintenance costs.

> Fast forward to now, and wow, have things changed. Git enjoys [up to 94% usage by developers](https://survey.stackoverflow.co/2022/#section-version-control-version-control-systems), and Subversion is a lot rarer in the wild than it used to be. Our traffic numbers inside GitHub back this up: fewer than 0.02% of requests made to the Git backend come through a Subversion endpoint, and only about 5,000 repositories see even a single Subversion request each month. It’s clear that Subversion support is no longer helping people migrate to Git.

> We will maintain Subversion support until January 8, 2024 on GitHub.com. After that date, it will be turned off and removed. Late in 2023, we’ll run a few hours-long and then day-long brownouts to help flush out any remaining use of the feature. 

https://github.blog/2023-01-20-sunsetting-subversion-support/

This is a bit of a nightmare. Let me try and recreate a least this latest version / tag for you with Git project sets and I'll think of what to do with older versions...

**Comment by dandar3** [comment]
@Bolfish 
Can you please try to import this version of the project set file using GitProvider please:

**_File > Import... > Team > Team Project Set > URL:_**
https://raw.githubusercontent.com/dandar3/android-google-play-services-ads/main/.projectset

I would recommend settings **_Settings > Preferences > Team > Git > Default repository folder_** to _**${workspace_loc}\git**_ to checkout the projects from GitHub inside the workspace. The SVN provider used to that without any setting.

**Comment by Bolfish** [comment]
@dandar3 
Do I need to install or upgrade my Eclipse, I got the following error:

```
The following errors occurred while importing projects. Some projects may not be loaded.
  Failed to read project description file from location 'C:\[Android]\eclipse\NewWorkspace\android-androidx-annotation-experimental\.project'.
    Failed to read project description file from location 'C:\[Android]\eclipse\NewWorkspace\android-androidx-annotation-experimental\.project'.
    C:\[Android]\eclipse\NewWorkspace\android-androidx-annotation-experimental\.project (The system cannot find the file specified.)
```

But the file is already on my hard drive.

dandar3 edited: display error statements as code to show paths (backslashes) correctly.

**Comment by dandar3** [comment]
@Bolfish 
I believe that is an issue with the Eclipse Git provider when you use `${workspace_loc}` as a default directory, I found that as an issue and that is why I suggested you to add another sub-directory to it, e.g. `${workspace_loc}\git` or whatever you want really as long as it's not the actual workspace directory. The error is a bit funny since it doesn't seem to write anything in there, it just fails with the first project in the project set.

You don't need to update unless you want to. This is the error with my Eclipse 4.8 (Photon) + Git Integration for Eclipse 5.0.0.201806131550-r which is similar to your, but yours threw me off as your paste was lacking some of the backslashes.

```
The following errors occurred while importing projects. Some projects may not be loaded.
  Invalid project description.
    Invalid project description.
      R:\Workspace\android-androidx-annotation-experimental overlaps the location of another project: 'android-androidx-annotation-experimental'
```

![image](https://github.com/dandar3/android-google-play-services-ads/assets/2507163/d901ae5c-c2f9-45de-88a4-389c0166f4ec)

In conclusion:
1. Navigate with a file manager to your new workspace location and remove any of the checked out projects (e.g. `android-androidx-annotation-experimental`)
2. Change `Default repository folder` to `${workspace_loc}` plus a subdirectory, e.g. `${workspace_loc}\github`
3. Try importing the project-set again.

![image](https://github.com/dandar3/android-google-play-services-ads/assets/2507163/8ff898a3-8efc-48ba-b05e-4cd14e3d949f)

**Comment by Bolfish** [comment]
@dandar3 
Finally I found out the reason was that my Eclipse version (Eclipse Juno) was too old.
After I finished my work, I had time to upgrade Eclipse and test again. Now everything is normal.
Thanks for your help.

**Comment by Bolfish** [comment]
@dandar3 

There seems to be something wrong with the library's dependent name.

![Image1](https://github.com/dandar3/android-google-play-services-ads/assets/156992199/fc77bc8e-c615-455e-9bb7-962c5b03c76e)

**Comment by dandar3** [comment]
@Bolfish 
Lovely, more troubles from the Eclipse EGit plugin. It checks out the libraries on disk with the actual repository name instead of the name of the project from `.project`, like Eclipse Subversion plugin used to do.

Some of the projects have the repository named with an `android-` prefix (e.g. `android-androidx-annotation`) while the actual project name was without (e.g. `androidx-annotation`) - https://github.com/dandar3/android-androidx-annotation/blob/main/.project

It's stupid and historical (wanted to show it was an Android project as opposed to other non-Android projects) and now it comes to bite me in the back. I'll need to update those as well for the new Git team projectset import... In the meantime can you please manually open the library `project.properties` and change from:

`android.library.reference.1=../androidx-[...]`
to
`android.library.reference.1=../android-androidx-[...]`

that should make them go back to green.

**Comment by dandar3** [comment]
No feedback since last commented, we assume the issue was resolved.
